### PR TITLE
test(daemon): deterministic fault-injection harness for connectivity

### PR DESCRIPTION
**Pull request by BunsDev**
First increment of `cave-58eoq.4` — cross-platform daemon connectivity fault injection.

## Why a harness rather than more per-test servers

Daemon connectivity tests each grew their own throwaway `createServer`, so the interesting failures — a hang, a half-written body, a reset mid-stream — were one-off and easy to get subtly wrong. Worse, an ad-hoc server that is hung or half-written is exactly the one whose `close()` never resolves, so a test asserting on a fault would itself hang. A suite that hangs on the failure path is a stalled required check, not coverage.

The harness owns two guarantees the individual tests could not:

- **Bounded** — every accepted socket is tracked and destroyed on `close()`. `server.close()` only stops new connections and waits for existing ones to drain; a deliberately hung response never drains, so without the registry the `hang` and `partial-response` cases leak a handle and hold the process open past the test.
- **Deterministic** — faults fire on *request count*, not wall clock, so `delayedReadyAfter: 3` behaves identically on a loaded CI runner and locally. That is what makes "bounded and non-flaky" hold rather than being an aspiration.

Nine modes: `healthy`, `delayed-ready`, `hang`, `partial-response`, `reset`, `crash-after`, `unauthorized`, `forbidden`, `oversized`.

## What the tests actually assert

The eleven tests drive the **real** `waitForDaemonReadiness` loop and the **real** `classifyDaemonFailureAvailability` classifier, rather than asserting the harness against itself. They pin behaviour worth pinning:

- a truncated body surfaces as a transport failure instead of parsing as an empty success;
- a reset carries no HTTP status, so it must not classify as `unhealthy`;
- 401/403 is `unauthorized` on a hub but `unhealthy` locally, where there is no credential to be wrong;
- repeated lifecycle churn leaks no sockets — the orphan check the acceptance criteria ask for.

Hang cases carry their own 150ms deadline: at the suite default they spent 6.2s asleep to prove what 632ms proves.

## Scope and deliberate omissions

Transport is loopback TCP. The HTTP contract is what the client code shares across platforms and loopback behaves identically on all three, so the same case list runs everywhere. Pipe- and permission-specific faults belong to `cave-58eoq.6`.

**Per-PR macOS coverage is deliberately not attempted.** The bead's acceptance says "all supported OSes", but macOS was removed from both PR matrices because hosted macOS minutes exhausted the org Actions limit and queued indefinitely, stalling every required check. Adding it back here would recreate that outage. Split out as `cave-5tpxy` for an owner decision; this runs on ubuntu-latest and windows-latest.

Sleep/wake, upgrade skew, and unusual-path cases remain for follow-up increments.

## Verification

`pnpm typecheck`, `pnpm lint`, and `check:tests-wired` (1597 files) all clean. The new test file is registered in `scripts/run-tests.mjs`. Local `pnpm test:app` was truncated by my own 900s timeout while inside `worktree-lifecycle-patrol.test.mjs` (patrol #97 at 829s — a pre-existing slow test that degrades as the checkout accumulates worktrees, currently ~29), so the full-suite verification here is CI's.

### CODE REVIEW

**Comment by BunsDev** [comment]
Reconciled the two autofix commits (`b8d540e0cc`, `c115413724`) with my own fix rather than force-pushing over them. Recording what was taken and what was declined, since we converged on the same finding from different directions.

**Taken — the docstring correction (`c115413724`).** I had written that `hang` *and* `partial-response` risk leaking a handle. Only `hang` does; the other destructive modes destroy their own socket. That was my error and the fix is right.

**Taken — dropping `sockets.clear()` (`b8d540e0cc`).** This was the substantive finding: `openSocketCount()` returned 0 unconditionally, so the leak assertions and `stressDaemonLifecycle`'s `leakedSockets` were true by construction.

**Declined — filtering `socket.destroyed` out of the count.** It would exclude a socket that was destroyed but never emitted `close` — a genuinely wedged handle — and report that as no leak. That reintroduces the original vacuousness one level down. The count stays the raw registry size.

**Added — `close()` now awaits each socket's real `close` event, bounded by a 250ms grace.** Removing `clear()` is necessary but not sufficient: `destroy()` is asynchronous, so a test reading `openSocketCount()` right after `await close()` could still see a non-zero count — the same assertion, flaky instead of vacuous. Awaiting fixes that; the grace keeps an unbounded wait from turning a wedged socket into a hung teardown, which is the failure this harness exists to prevent. On timeout the socket stays in the registry and surfaces as a non-zero count.

Also added a test pinning the count non-zero while a connection is open and zero after close. To be precise: it would not by itself have caught the original bug — both assertions held under it too. It guards against a regression to a stub; what makes the zero meaningful is the awaited close.

12 tests pass in 839ms, `pnpm typecheck` and `pnpm lint` clean.

File: src/lib/daemon-fault-harness.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+ * resolves, so a test asserting on a fault would itself hang. A suite that
+ * hangs on the failure path is not coverage; it is a stalled required check.
+ *
+ * So this harness owns two guarantees the individual tests could not:
+ *
+ *  - **Bounded.** Every accepted socket is tracked and destroyed on `close()`.
+ *    Node's `server.close()` only stops new connections and waits for existing
+ *    ones to drain; a deliberately hung response never drains. Without the
+ *    socket registry, `hang` and `slow-body` would leak a handle and hold the
+ *    process open past the test.
```
[the remark this answers is not in the document]
Comment (BunsDev) [reply src/lib/daemon-fault-harness.ts]: Fixed in c115413724 and retained through 5499e5ebfb: the module documentation now names the implemented partial-response mode rather than slow-body.
